src/persist-client/src/internal/machine.rs

```diff
@@ -111,7 +111,17 @@ where
     }
 
     pub async fn add_rollup_for_current_seqno(&mut self) -> RoutineMaintenance {
-        let rollup = self.applier.write_rollup_blob(&RollupId::new()).await;
+        let state = self.applier.clone_state_for_rollup();
```

Comment:
I'll play with this some more, I don't love the current model either. 

The two challenges were:
1) We need a static state to fetch diffs for, otherwise we'd be chasing our tail trying to get the diffs for the latest seqno that keeps advancing
2) I wanted to avoid encoding the proto within the state RwLock

I'll riff on `encode_rollup_blob_without_diffs` and see where it ends up

Reply:
why (2)? I wouldn't be terribly surprised if encoding it to a proto is faster than cloning it

anyway, sorry if this wasn't clear, but I talked myself out of the `encode_rollup_blob_without_diffs` suggestion by the time i finished typing it out. my concrete proposal was a new method on Applier that 1) does the work that `clone_state_for_rollup` is doing and then 2) calls machine.write_rollup_for_state for you. then remove `clone_state_for_rollup` from Applier